engine/apps/mobile_app/views.py

```diff
@@ -217,13 +228,14 @@ def _proxy_request(self, request: Request, *args, **kwargs) -> Response:
                 downstream_url,
                 data=request.body,
                 params=request.query_params.dict(),
-                headers=self._get_downstream_headers(request, user),
+                headers=self._get_downstream_headers(request, downstream_backend, user),
             )
 
             return Response(status=downstream_response.status_code, data=downstream_response.json())
         except (
             requests.exceptions.RequestException,
             requests.exceptions.JSONDecodeError,
+            CloudAuthApiException,
```

Comment:
if the cloud auth API's `POST /v1/sign` endpoint returns anything that is non-200, we return HTTP 502 to the client (aka the mobile app)